File: School_Food_iOS/School_Food/Extension/DateExtension.swift
//
//  DateExtension.swift
//  School_Food
//
//  Created by 박진 on 2020/08/25.
//  Copyright © 2020 com.meals.school_food. All rights reserved.
//

import Foundation

extension Date {
    
    func yearDateFormat() -> String {
        let format = DateFormatter()
        format.dateFormat = "yyyyMMdd"
        return format.string(from: self)
    }
    
    func monthDateFormat() -> String {
        let format = DateFormatter()
        format.dateFormat = "yyyyMM"
        return format.string(from: self)
    }
    
    func getDateFormat() -> Date {
        let format = DateFormatter()
        format.dateFormat = "HH:mm"
        return format.date(from: format.string(from: self))!
    }
}

extension String {
    
    func getTime() -> Date {
        let format = DateFormatter()
        format.dateFormat = "HH:mm"
        return format.date(from: self)!
    }
    
    func getDateFormat() -> String {
        let beforeFormat = DateFormatter()
        beforeFormat.dateFormat = "yyyyMMdd"
        
        let afterFormat = DateFormatter()
        afterFormat.dateFormat = "yyyy년 M월 d일"
        
        let date = beforeFormat.date(from: self)!
        return afterFormat.string(from: date)
    }
}

File: School_Food_Android/presentation/src/main/java/com/meals/school_food/widget/extension/DateExtension.kt
package com.meals.school_food.widget.extension

import java.text.SimpleDateFormat
import java.util.*

fun String.getTime(): Date {
    val format = SimpleDateFormat("HH:mm", Locale.getDefault())
    return format.parse(this)!!
}

fun Date.getFormatDate(): Date {
    val format = SimpleDateFormat("HH:mm", Locale.getDefault())
    return format.parse(format.format(this))!!
}

fun Date.krDateFormat() : String {
    val format = SimpleDateFormat("yyyy년 M월 d일", Locale.getDefault())
    return format.format(this)
}

fun Date.monthDateFormat() : String {
    val format = SimpleDateFormat("yyyyMM", Locale.getDefault())
    return format.format(this)
}

fun Date.dayDateFormat() : String {
    val format = SimpleDateFormat("yyyyMMdd", Locale.getDefault())
    return format.format(this)
}

fun String.getDateFormat() : String {
    val beforeFormat = SimpleDateFormat("yyyy년 MM월 dd일", Locale.getDefault())
    val afterFormat = SimpleDateFormat("yyyyMMdd", Locale.getDefault())
    val date = beforeFormat.parse(this)!!
    return afterFormat.format(date)
}

fun String.getDateFormat2() : String {
    val beforeFormat = SimpleDateFormat("yyyyMMdd", Locale.getDefault())
    val afterFormat = SimpleDateFormat("yyyyMMdd", Locale.getDefault())
    val date = beforeFormat.parse(this)!!
    return afterFormat.format(date)
}